packages/x-charts/src/LineChart/MarkPlot.tsx

```diff
@@ -76,102 +80,111 @@ function MarkPlot(props: MarkPlotProps) {
   const defaultXAxisId = xAxisIds[0];
   const defaultYAxisId = yAxisIds[0];
 
-  return (
-    <g {...other}>
-      {stackingGroups.flatMap(({ ids: groupIds }) => {
-        return groupIds.map((seriesId) => {
-          const {
-            xAxisId = defaultXAxisId,
-            yAxisId = defaultYAxisId,
-            stackedData,
-            data,
-            showMark = true,
-            shape = 'circle',
-          } = series[seriesId];
-
-          if (showMark === false) {
-            return null;
-          }
-
-          const xScale = getValueToPositionMapper(xAxis[xAxisId].scale);
-          const yScale = yAxis[yAxisId].scale;
-          const xData = xAxis[xAxisId].data;
-
-          if (xData === undefined) {
-            throw new Error(
-              `MUI X: ${
-                xAxisId === DEFAULT_X_AXIS_KEY
-                  ? 'The first `xAxis`'
-                  : `The x-axis with id "${xAxisId}"`
-              } should have data property to be able to display a line plot.`,
+  const children = [];
+
+  for (const stackingGroup of stackingGroups) {
+    const groupIds = stackingGroup.ids;
+
+    for (const seriesId of groupIds) {
+      const {
+        xAxisId = defaultXAxisId,
+        yAxisId = defaultYAxisId,
+        stackedData,
+        data,
+        showMark = true,
+        shape = 'circle',
+      } = series[seriesId];
+
+      if (showMark === false) {
+        continue;
+      }
+
+      const xScale = getValueToPositionMapper(xAxis[xAxisId].scale);
+      const yScale = yAxis[yAxisId].scale;
+      const xData = xAxis[xAxisId].data;
+
+      if (xData === undefined) {
+        throw new Error(
+          `MUI X: ${
```

Comment:
Not directly related, but I see this in the diff https://github.com/mui/mui-public/issues/289

```suggestion
          `MUI X Charts: ${
```

Reply:
Handled this in https://github.com/mui/mui-x/pull/17547